lib/active_set/filtering/operation.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+require_relative '../attribute_instruction'
+
+class ActiveSet
+  module Filtering
+    class Operation
+      def initialize(set, instructions_hash)
+        @set = set
+        @instructions_hash = instructions_hash
+      end
+
+      def execute
+        attribute_instructions = @instructions_hash
+                                   .flatten_keys
+                                   .map { |k, v| AttributeInstruction.new(k, v) }
+
+        activerecord_filtered_set = attribute_instructions.reduce(@set) do |set, attribute_instruction|
+          maybe_set_or_false = ActiveRecordStrategy.new(set, attribute_instruction).execute
+          next set unless maybe_set_or_false
+
+          attribute_instruction.processed = true
+          maybe_set_or_false
+        end
+
+        return activerecord_filtered_set if attribute_instructions.all?(&:processed?)
+
+        attribute_instructions.reject(&:processed?).reduce(activerecord_filtered_set) do |set, attribute_instruction|
+          maybe_set_or_false = EnumerableStrategy.new(set, attribute_instruction).execute
+          maybe_set_or_false.presence || set
+        end
+      end
+
+      def operation_instructions
+        @instructions_hash.symbolize_keys
+      end
+    end
+
+    class EnumerableStrategy
+      def initialize(set, attribute_instruction)
+        @set = set
+        @attribute_instruction = attribute_instruction
+      end
+
+      def execute
+        return false unless @set.respond_to? :select
+
+        @set.select do |item|
+          if execute_attribute_comparison_for?(item)
+            next attribute_comparison_for(item)
+          elsif execute_class_method_call_for?(item)
+            next class_method_call_for(item)
+          else
+            next false
+          end
+        end
+      end
+
+      private
+
+      def execute_attribute_comparison_for?(item)
+        attribute_item = attribute_item_for(item)
+
+        return false unless attribute_item
+        return false unless attribute_item.respond_to?(@attribute_instruction.attribute)
+        return false unless attribute_item.method(@attribute_instruction.attribute).arity <= 0
```

Comment:
Directly defined instance method getters with have an arity of zero; getters defined thru meta-programming (like using `method_missing`) will have an arity of -1.